### DISCUSSION TITLE
RUST Web API to run YoloV5 inferences

### DISCUSSION THREAD

**Discussion by masc-it** [ultralytics/yolov5#8681] [Show and tell]
Hello everybody,
I'm Mauro, 24yo from Italy and I'm currently studing AI (MSc).

In the last few weeks I have been tinkering with Rust and I thought that a "real" project could be the best way to learn the language (well... scratching the surface). 
Since I'm a Computer Vision enthusiast and I've been using YoloV5 for a long while, I have decided to make a simple program in Rust to run inferences with YoloV5, using the [opencv crate](https://github.com/twistedfall/opencv-rust) . 

Actually, on top of the prediction modules I've also built a web API using [actix-web](https://github.com/actix/actix-web). With the **/predict** endpoint you can POST an image, run the predictions and get back the results (image with boxes or JSON).

The code is quite modular, so you can run inferences indipendently from the web-server, if necessary.

_Please do not judge the level of the code, I'm just getting started with Rust :)_

Anyways, I hope you'll appreciate it!

The API is also available as Docker Image, with everything properly setup (Rust, OpenCV, etc) in a Dockerfile.
Check out the project: [yolov5-api-rust](https://github.com/masc-it/yolov5-api-rust).
[Docker Hub](https://hub.docker.com/r/mascit/rust-yolov5-api)

**Comment by AyushExel** [comment]
It a cool project! Thanks for sharing
